src/main/java/com/benmorant/game/holycraap/model/dao/ItemDao.java

```diff
@@ -1,9 +1,12 @@
 package com.benmorant.game.holycraap.model.dao;
 
 import com.benmorant.game.holycraap.model.entity.Item;
+import java.util.Optional;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
 
+@Repository
 public interface ItemDao extends JpaRepository<Item, Integer> {
 
-  Item findByName(String name);
+  Optional<Item> findByName(String name);
```

Comment:
Is it an item or item class/type? If 10 people have a "Mithril Sword" wouldn't they all have the same name?

Reply:
yes, indeed. But what would happen if the given String doesn't appear to be found as an Item's name ? NPE right ? I thought Optional<Item> would be a solution to avoid that

Reply:
That's not what I'm talking about.

The question is: what does an `Item` represent here? Are all "Mithril Sword" in the game one `Item` or are they different `Item`?

Reply:
mmm... different items, I guess.

Reply:
Given it's an adventure game and not a role playing game, it's actually OK.